### PR TITLE
Excalibur dev

### PR DESCRIPTION
**Pull request by ajgdls**
Before they diverge by too much.  This is work in progress so review it at your own risk...

### CODE REVIEW

**Comment by ajgdls** [comment]
@GDYendell

File: odinApp/src/odinDetector.cpp [hunk 1]
```diff
@@ -226,18 +239,49 @@ int OdinDetector::createOdinDataParams()
...
+        setStringParam(ADStatusMessage, "Acquiring...");
+      }
+      callParamCallbacks();
+    }
+
+    // Check for any errors
+    // If we are connected to the server then all errors are generated by
+    // the server itself.
+    std::string error_msg = "";
+    mErrorMessage->get(error_msg);
```
Comment (GDYendell) [inline odinApp/src/odinDetector.cpp:281]: The IOC crashes here when I enable status polling. It looks like mErrorMessage is not created.

**Comment by ajgdls** [comment]
The only conclusion that I can find is that I've squashed the code when I rebased.  Another reason to split these parts out so re-bases don't keep affecting the same areas of code...
I'll re-implement it now and push the update.

**Comment by ajgdls** [comment]
@GDYendell I have pushed a fix but I can't test it as the crash still occurs until I get your latest changes.

**Comment by GDYendell** [comment]
@ajgdls Try it with that new commit. I can now run it with no server and enable status polling and it doesn't crash.

**Comment by ajgdls** [comment]
@GDYendell yes that works for me now too.  Thanks.

**Pull request merged**